src/util.ts

```diff
@@ -2002,21 +2002,24 @@ export class Util {
         let parentTypeName = '';
         let errorRange: Range;
         let containsDynamic = false;
+        let continueEverything = true;
         for (let i = 0; i < typeChain.length; i++) {
             const chainItem = typeChain[i];
             const dotSep = chainItem.separatorToken?.text ?? '.';
             if (i > 0) {
                 fullChainName += dotSep;
             }
             fullChainName += chainItem.name;
-            parentTypeName = previousTypeName;
-            fullErrorName = previousTypeName ? `${previousTypeName}${dotSep}${chainItem.name}` : chainItem.name;
-            previousTypeName = chainItem.type?.toString() ?? '';
-            itemName = chainItem.name;
-            containsDynamic = containsDynamic || (isDynamicType(chainItem.type) && !isAnyReferenceType(chainItem.type));
-            if (!chainItem.isResolved) {
-                errorRange = chainItem.range;
-                break;
+            if (continueEverything) {
+                parentTypeName = previousTypeName;
+                fullErrorName = previousTypeName ? `${previousTypeName}${dotSep}${chainItem.name}` : chainItem.name;
+                previousTypeName = chainItem.type?.toString() ?? '';
+                itemName = chainItem.name;
+                containsDynamic = containsDynamic || (isDynamicType(chainItem.type) && !isAnyReferenceType(chainItem.type));
+                if (!chainItem.isResolved) {
+                    errorRange = chainItem.range;
+                    continueEverything = false;
```

Comment:
It seems that you want to escape the loop if you get to this point. Wouldn't a `break` solve this instead of using a variable?

Reply:
I still want the "fullChainName"... just not the other stuff..
I changed the name of this variable to reflect better what it means.